frontend/src/components/AddNames.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState, useContext } from "react";
+import { GlobalContext } from "../context/GlobalContext";
+
+function AddNames() {
```

Comment:
I'm not sure why you have this as a function and not a component. Looking down through the file, it seems like you made most of the files this way. I think your syntax is probably equivalent syntax to the component syntax? But it doesn't read as nicely IMO. React might be using functions everywhere under the hood, but it's nice to be able to separate things so that functions can be operative pieces of code that make something happen, while components are pieces of the display, essentially serving to create the DOM.
```suggestion
class NameInputBox extends Component {
```